src/main/java/org/sonarsource/sonarlint/ls/log/LanguageClientLogger.java

```diff
@@ -71,68 +71,81 @@ public void initialize(boolean showVerboseLogs) {
     this.showVerboseLogs = showVerboseLogs;
   }
 
-  private void log(String prefix, String formattedMessage, boolean isDebugOrTrace, boolean isFromAnalysis) {
-    if ((!isFromAnalysis || showAnalyzerLogs) && (showVerboseLogs || !isDebugOrTrace)) {
+  public void log(String formattedMessage, LogLevel level) {
+    log(getLogPrefix(level), formattedMessage, isDebugOrTrace(level));
+  }
+
+  private static boolean isDebugOrTrace(LogLevel level) {
+    return level == LogLevel.DEBUG || level == LogLevel.TRACE;
+  }
+
+  private static String getLogPrefix(LogLevel level) {
+    return switch (level) {
+      case DEBUG -> "Debug";
+      case TRACE -> "Trace";
+      case INFO -> "Info";
+      case WARN -> "Warn";
+      case ERROR -> "Error";
+    };
+  }
+
+  private void log(String prefix, String formattedMessage, boolean isDebugOrTrace) {
+    if (showVerboseLogs || !isDebugOrTrace) {
       client.logMessage(new MessageParams(MessageType.Log, prefix(prefix, formattedMessage)));
     }
   }
 
   @Override
   public void onChange(@CheckForNull WorkspaceSettings oldValue, WorkspaceSettings newValue) {
     this.showVerboseLogs = newValue.showVerboseLogs();
-    this.showAnalyzerLogs = newValue.showAnalyzerLogs();
   }
 
   private String prefix(String prefix, String formattedMessage) {
     return "[" + prefix + " - " + LocalTime.now(clock).format(LOG_DATE_FORMAT) + "] " + formattedMessage;
   }
 
-  void error(String formattedMessage, boolean isFromAnalysis) {
-    log("Error", formattedMessage, false, isFromAnalysis);
-  }
-
   public void error(String formattedMessage) {
-    error(formattedMessage, false);
+    log("Error", formattedMessage, false);
   }
 
-  public void error(String formattedMessage, Throwable t) {
+  public void errorWithStackTrace(String formattedMessage, Throwable t) {
     var sw = new StringWriter();
     var pw = new PrintWriter(sw);
     t.printStackTrace(pw);
     var sStackTrace = sw.toString();
-    error(formattedMessage + "\n" + sStackTrace, false);
-  }
-
-  void warn(String formattedMessage, boolean isFromAnalysis) {
-    log("Warn ", formattedMessage, false, isFromAnalysis);
+    error(formattedMessage + "\n" + sStackTrace);
   }
 
   public void warn(String formattedMessage) {
-    warn(formattedMessage, false);
+    log("Warn", formattedMessage, false);
   }
 
-  void info(String formattedMessage, boolean isFromAnalysis) {
-    log("Info ", formattedMessage, false, isFromAnalysis);
+  public void warnWithStackTrace(String formattedMessage, Throwable t) {
+    var sw = new StringWriter();
+    var pw = new PrintWriter(sw);
+    t.printStackTrace(pw);
+    var sStackTrace = sw.toString();
```

Comment:
IMHO, this should be extracted to a method shared with the other `*WithStackTrace` methods.